test/example.c

```diff
@@ -138,7 +142,11 @@ void test_gzio(const char *fname, unsigned char *uncompr, z_size_t uncomprLen) {
     }
     /* Read uncompressed data - hello, hello! string twice */
     strcpy((char*)uncompr, "garbages");
-    if (PREFIX(gzread)(file, uncompr, (unsigned)uncomprLen) != (int)(len + len)) {
+    read = PREFIX(gzread)(file, uncompr, (unsigned)uncomprLen);
+    fprintf(stderr, "read %d\n", read);
+    if (read != (int)(len + len)) {
+        read =  PREFIX(gzread)(file, uncompr, (unsigned)uncomprLen);
```

Comment:
Should print contents of "uncompr" as hexadecimal before attempting to read more bytes...